src/main/webapp/filter_entry.js

```diff
@@ -0,0 +1,105 @@
+  constructor(parentFilterArea) { 
+  
+    this.activeFilterArea = parentFilterArea;
+
+    /* filterEntryArea is displayed when the user clicks add filter. It first asks for a type of filter, 
+       then asks the user to enter the keyword for that filter type */
+    this.filterEntryArea = document.createElement("div");
+    this.filterEntryArea.setAttribute("class", "filter-tag-area");
+    this.filterEntryArea.setAttribute("id", "filter-entry");
+
+    /* The user selects the type of property they want to filter by in filterFieldInput*/
+    this.filterFieldInput = document.createElement("input");
+    this.filterFieldInput.setAttribute("list", "filter-datalist");
+    this.filterFieldInput.setAttribute("class", "filter-input-area");
+    this.filterFieldInput.setAttribute("placeholder", "Filter by:");
+    this.filterFieldInput.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        this.filterParamLabel.textContent = `${this.filterFieldInput.value}:`;
+        this.filterFieldInput.removeChild(this.filterDataList);
+        this.filterEntryArea.removeChild(this.filterFieldInput);
+        this.filterEntryArea.appendChild(this.filterParamInput);
+        this.filterEntryArea.appendChild(this.filterParamLabel);
+      }
+    });
+
+    this.filterDataList = document.createElement("datalist");
+    this.filterDataList.setAttribute("id", "filter-datalist");
+    this.optionMap = new Map();
+    this.optionMap.set("Organization Name", "orgNames");
+    this.optionMap.set("Address", "orgStreetAddresses");
+    this.optionMap.set("Available Resources", "resourceCategories");
+    for (const optionKey of this.optionMap.keys()) {
+      const option = document.createElement("option");
+      option.value = optionKey;
+      this.filterDataList.appendChild(option);
+    }
+
+    /* After the type has been chosen, the actual filter param is entered here */
+    this.filterParamInput = document.createElement("input");
+    this.filterParamInput.setAttribute("class", "filter-input-area");
+    this.filterParamLabel = document.createElement("label");
+    this.filterParamLabel.setAttribute("class", "filter-tag-label");
+    this.filterParamInput.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        /* If the user enters an unsupported type, it is ignored on the servlet side */
+        console.log(this.optionMap.get(this.filterFieldInput.value));
+        this.filterEntryArea.dispatchEvent(new CustomEvent('onParamEntry', {
+          bubbles: true,
+          detail : {
+            urlParamKey: this.optionMap.get(this.filterFieldInput.value),
+            urlParamValue: this.filterParamInput.value
+          }
+        }));
+        this.filterParamInput.value = "";
+        this.filterFieldInput.value = "";
+        this.filterEntryArea.removeChild(this.filterParamInput);
+        this.filterEntryArea.removeChild(this.filterParamLabel);
+        this.filterEntryArea.dispatchEvent(this.closeEvent);
```

Comment:
instead of this.closeEvent, just create a new one each time 

new Event('onRemove')